xradar/accessors.py

```diff
@@ -53,3 +58,47 @@ def create_xradar_dataarray_accessor(name, funcs):
     cls_name = "".join([name.capitalize(), "Accessor"])
     accessor = type(cls_name, (object,), methods)
     return xr.register_dataarray_accessor(name)(accessor)
+
+
+class XradarAccessor:
+    """
+    Common Datatree, Dataset, DataArray accessor functionality.
+    """
+
+    def __init__(
+        self, xarray_obj: xr.Dataset | xr.DataArray | dt.DataTree
+    ) -> XradarAccessor:
+        self.xarray_obj = xarray_obj
+
+
+@xr.register_dataarray_accessor("xradar")
+class XradarDataArrayAccessor(XradarAccessor):
+    """Adds a number of xradar specific methods to xarray.DataArray objects."""
+
+    def georeference(self) -> xr.DataArray:
```

Comment:
Would be great to have the opportunity here to provide keywords for the georeferencing (ke, earth_radius etc).

Reply:
Good point - I can add that in here.